### DISCUSSION TITLE
Juggluco (EverSence) -> xDrip(640G/EverSence) -> Nightscout

### DISCUSSION THREAD

**Discussion by Geniy83** [NightscoutFoundation/xDrip#3619] [Show and tell]
I made a separate build of xDrip+ for myself (not for public use). There is a problem - when you add calibration to Nightscout, uncalibrated data is lost. Tried using your latest version of xDrip (20240723) - same problem. Please tell me where in the code I can correct this point so that calibrated data is sent to Nightscout?
When doing calibration I use the plugin: xDrip Original - classic algorithm, slider enabled - Use glucose from plugin.

**Comment by Navid200** [comment]
We avoid adding calibration options that way.
Calibration should be done by the collector.  In the scenario you are describing, calibration should be done on the Libre device, not in xDrip.
This is a safety issue.

**Reply by Geniy83** [reply]
I use this assembly myself; the assembly is not public. I didn't find this in the code myself. Can you tell me where I can change this?

**Reply by Navid200** [reply]
No, I have never looked into that.

**Comment by Geniy83** [comment]
I use a Sibionics sensor with Juggluco via EverSence data source. Juggluco cannot enter calibrations, but I need them. I calibrate data in xDrip and want to see calibrated data in Nightscout and in the history for statistics. But on the EverSence data source this is impossible. There is also an application like Poctech, which works with xDrip only via EverSence. What to do in this situation?

**Reply by Navid200** [reply]
Calibration is done in xDrip only if there is no other way to calibrate.
Ideally, calibration is done by the device itself.  For example, if you use Dexcom devices, you calibrate by submitting the calibration to the Dexcom transmitter.  It's the Dexcom transmitter that does the calibration.  In that case, if you decide to calibrate yourself inside xDrip, you cannot.  xDrip does not offer that option.  

How do people who use the Sibionics sensor and do not use xDrip calibrate them?

**Reply by Geniy83** [reply]
Native applications sibionics and poctech use, in my opinion, a bad calibration algorithm. Today I consider the calibration algorithm used in xDrip, the most ideal. And I would like to apply it to raw data received from EverSense

**Reply by Geniy83** [reply]
And Juggluco doesn't use calibrations at all

**Reply by Navid200** [reply]
But, that is not the purpose of xDrip to go head to head against the manufacturer of a sensor.
If a sensor manufacturer has gone through the trouble of getting it approved by the authorities for distribution, they must know how it should be calibrated better than anyone else.

Have you reached out to the manufacturer?  What do they say in response to your criticism of their calibration scheme?

**Reply by Geniy83** [reply]
> Обращались ли вы к производителю? Что они говорят в ответ на вашу критику их схемы калибровки?

Обращался только к poctech и был проигнорирован)
